### PR TITLE
SslServerAuthenticationOptions - update documentation

### PR DESCRIPTION
**Pull request by aik-jahoda**
## Summary

Collect all changes from #3916 related to SslServerAuthenticationOptions

### CODE REVIEW

File: xml/System.Net.Security/SslServerAuthenticationOptions.xml [hunk 1]
```diff
@@ -314,7 +334,7 @@
         <ReturnType>System.Net.Security.ServerCertificateSelectionCallback</ReturnType>
       </ReturnValue>
       <Docs>
-        <summary>To be added.</summary>
+        <summary>Gets or sets the <see cref="T:System.Net.Security.ServerCertificateSelectionCallback" /> delegate responsible for the server certificate selection.</summary>
```
Comment (ManickaP) [inline xml/System.Net.Security/SslServerAuthenticationOptions.xml:337]: There is a nice [comment](https://github.com/dotnet/runtime/blob/master/src/libraries/System.Net.Security/src/System/Net/Security/SecureChannel.cs#L636-L639) explaining the server certificate selection. Maybe it's worth mentioning.

**Comment by ManickaP** [comment]
I tried to look into usages of this class to add deeper documentation for the properties. Though, it might not be something desired here since it's API reference docs and also the class might potentially be used in different scenarios/contexts than `SslStream`. So please take my comments with that in mind.

File: xml/System.Net.Security/SslServerAuthenticationOptions.xml [hunk 2]
```diff
@@ -126,9 +133,20 @@
...
-        <remarks>To be added.</remarks>
+        <summary>Gets or sets the certificate revocation mode for certificate validation.</summary>
+        <value>One of the values in <see cref="T:System.Security.Cryptography.X509Certificates.X509RevocationMode" />: <see langword="NoCheck" />, <see langword="Online" />, or <see langword="Offline" />. The default is <see langword="Online" />.</value>       
+  <remarks>
+          <format type="text/markdown"><![CDATA[  
+  
+## Remarks  
+ When using certificates, the system validates that the client certificate is not revoked, by checking that the client certificate is not in the revoked certificate list. This check can be performed either by checking online or by checking against a cached revocation list. Revocation checking can be turned off by setting this property to <xref:System.Security.Cryptography.X509Certificates.X509RevocationMode.NoCheck>.  
+  
+ For more information, see [Working with Certificates](/dotnet/framework/wcf/feature-details/working-with-certificates).
```
[the remark this answers is not in the document]
Comment (aik-jahoda) [reply xml/System.Net.Security/SslServerAuthenticationOptions.xml:144]: this reference is used on few places in the file already

File: xml/System.Net.Security/SslServerAuthenticationOptions.xml [hunk 3]
```diff
@@ -179,7 +197,7 @@
         <ReturnType>System.Boolean</ReturnType>
       </ReturnValue>
       <Docs>
-        <summary>To be added.</summary>
+        <summary>Gets or sets a <see cref="T:System.Boolean" /> value that specifies whether the client is asked for a certificate for authentication. Note that this is only a request -- if no certificate is provided, the server still accepts the connection request.</summary>
```
[the remark this answers is not in the document]
Comment (aik-jahoda) [reply xml/System.Net.Security/SslServerAuthenticationOptions.xml:200]: I borrowed it from there: https://review.docs.microsoft.com/en-us/dotnet/api/system.net.security.sslstream.authenticateasserverasync?view=netcore-1.1&branch=pr-en-us-4350#System_Net_Security_SslStream_AuthenticateAsServerAsync_System_Security_Cryptography_X509Certificates_X509Certificate_System_Boolean_System_Security_Authentication_SslProtocols_System_Boolean_ . The enabledSslProtocols parameter has the same meaning as SslServerAuthenticationOptions.ClientCertificateRequired in overloaded AuthenticateAsServerXXX

File: xml/System.Net.Security/SslServerAuthenticationOptions.xml [hunk 4]
```diff
@@ -206,8 +224,8 @@
         <ReturnType>System.Security.Authentication.SslProtocols</ReturnType>
       </ReturnValue>
       <Docs>
-        <summary>To be added.</summary>
-        <value>To be added.</value>
+        <summary>Gets or sets the <see cref="T:System.Security.Authentication.SslProtocols" /> value that represents the protocol used for authentication.</summary>
```
[the remark this answers is not in the document]
Comment (aik-jahoda) [reply xml/System.Net.Security/SslServerAuthenticationOptions.xml:227]: Fixed in SslStream too. 
Added remark with the logic into the SslProtocols enum.

File: xml/System.Net.Security/SslServerAuthenticationOptions.xml [hunk 5]
```diff
@@ -99,7 +106,7 @@
         <ReturnType>System.Collections.Generic.List&lt;System.Net.Security.SslApplicationProtocol&gt;</ReturnType>
       </ReturnValue>
       <Docs>
-        <summary>To be added.</summary>
+        <summary>Gets or sets a list of <see cref="T:System.Net.Security.SslApplicationProtocol" />.</summary>
         <value>To be added.</value>
         <remarks>To be added.</remarks>
```
[the remark this answers is not in the document]
Comment (aik-jahoda) [reply xml/System.Net.Security/SslServerAuthenticationOptions.xml:111]: PR documenting the application protocol: https://github.com/dotnet/dotnet-api-docs/pull/4350

File: xml/System.Net.Security/SslServerAuthenticationOptions.xml [hunk 6]
```diff
@@ -314,9 +334,16 @@
...
       <Docs>
-        <summary>To be added.</summary>
+        <summary>Gets or sets the <see cref="T:System.Net.Security.ServerCertificateSelectionCallback" /> delegate responsible for the server certificate selection.</summary>
         <value>To be added.</value>
-        <remarks>To be added.</remarks>
+        <remarks>
+          <format type="text/markdown"><![CDATA[  
+  
+## Remarks         
+        During the server connection establishment, the server first uses <xref:System.Net.Security.SslServerAuthenticationOptions.ServerCertificateSelectionCallback> callback. If the delegate is not provided server uses  <xref:System.Net.Security.LocalCertificateSelectionCallback> provided as argument of <xref:System.Net.Security.SslStream.%23ctor%2a?displayProperty=nameWithType> (for the .NET Framework compatibility). If any of those callbacks are provided, the server falls back to <xref:System.Net.Security.SslServerAuthenticationOptions.ServerCertificate>.
```
Comment (ManickaP) [inline xml/System.Net.Security/SslServerAuthenticationOptions.xml:343]: any of those callbacks are provided -->  none/neither of those callback are provided

What you're writing has opposite meaning.

File: xml/System.Security.Authentication/SslProtocols.xml [hunk 7]
```diff
@@ -39,6 +39,10 @@
     <summary>Defines the possible versions of <see cref="T:System.Security.Authentication.SslProtocols" />.</summary>
     <remarks>
       <format type="text/markdown"><![CDATA[  
+      
+## Remarks
+
+  During the handshake, the client sends a list of SSL versions and the server chooses from that list the best matching version.
```
Comment (ManickaP) [inline xml/System.Security.Authentication/SslProtocols.xml:45]: NIT: 
the server chooses from that list the best matching version.  --> the server chooses the best matching version from that list.

**Pull request merged**